pkg/goanalysis/runner_checker.go

```diff
@@ -94,106 +104,140 @@ func (act *action) analyze() {
+	if mod := act.Package.Module; mod != nil {
 		module.Path = mod.Path
 		module.Version = mod.Version
 		module.GoVersion = mod.GoVersion
 	}
 
 	// Run the analysis.
 	pass := &analysis.Pass{
-		Analyzer:     act.a,
-		Fset:         act.pkg.Fset,
-		Files:        act.pkg.Syntax,
-		OtherFiles:   act.pkg.OtherFiles,
-		IgnoredFiles: act.pkg.IgnoredFiles,
-		Pkg:          act.pkg.Types,
-		TypesInfo:    act.pkg.TypesInfo,
-		TypesSizes:   act.pkg.TypesSizes,
-		TypeErrors:   act.pkg.TypeErrors,
+		Analyzer:     act.Analyzer,
+		Fset:         act.Package.Fset,
+		Files:        act.Package.Syntax,
+		OtherFiles:   act.Package.OtherFiles,
+		IgnoredFiles: act.Package.IgnoredFiles,
+		Pkg:          act.Package.Types,
+		TypesInfo:    act.Package.TypesInfo,
+		TypesSizes:   act.Package.TypesSizes,
+		TypeErrors:   act.Package.TypeErrors,
 		Module:       module,
 
 		ResultOf:          inputs,
-		Report:            func(d analysis.Diagnostic) { act.diagnostics = append(act.diagnostics, d) },
-		ImportObjectFact:  act.importObjectFact,
+		Report:            func(d analysis.Diagnostic) { act.Diagnostics = append(act.Diagnostics, d) },
+		ImportObjectFact:  act.ObjectFact,
 		ExportObjectFact:  act.exportObjectFact,
-		ImportPackageFact: act.importPackageFact,
+		ImportPackageFact: act.PackageFact,
 		ExportPackageFact: act.exportPackageFact,
-		AllObjectFacts:    act.allObjectFacts,
-		AllPackageFacts:   act.allPackageFacts,
+		AllObjectFacts:    act.AllObjectFacts,
+		AllPackageFacts:   act.AllPackageFacts,
 	}
-
+	pass.ReadFile = analysisinternal.MakeReadFile(pass)
 	act.pass = pass
-	act.r.passToPkgGuard.Lock()
-	act.r.passToPkg[pass] = act.pkg
-	act.r.passToPkgGuard.Unlock()
 
-	if act.pkg.IllTyped {
+	act.runner.passToPkgGuard.Lock()
+	act.runner.passToPkg[pass] = act.Package
+	act.runner.passToPkgGuard.Unlock()
+
+	act.Result, act.Err = func() (any, error) {
+		// NOTE(golangci-lint):
 		// It looks like there should be !pass.Analyzer.RunDespiteErrors
-		// but govet's cgocall crashes on it. Govet itself contains !pass.Analyzer.RunDespiteErrors condition here,
+		// but govet's cgocall crashes on it.
+		// Govet itself contains !pass.Analyzer.RunDespiteErrors condition here,
 		// but it exits before it if packages.Load have failed.
-		act.err = fmt.Errorf("analysis skipped: %w", &pkgerrors.IllTypedError{Pkg: act.pkg})
-	} else {
-		startedAt = time.Now()
+		if act.Package.IllTyped {
+			return nil, fmt.Errorf("analysis skipped: %w", &pkgerrors.IllTypedError{Pkg: act.Package})
+		}
+
+		t1 := time.Now()
 
-		act.result, act.err = pass.Analyzer.Run(pass)
+		result, err := pass.Analyzer.Run(pass)
+		if err != nil {
+			return nil, err
+		}
 
-		analyzedIn := time.Since(startedAt)
-		if analyzedIn > time.Millisecond*10 {
+		analyzedIn := time.Since(t1)
+		if analyzedIn > 10*time.Millisecond {
 			debugf("%s: run analyzer in %s", act, analyzedIn)
 		}
-	}
 
-	// disallow calls after Run
+		// correct result type?
```

Comment:
What is the question here? Why they're different? The code seems to properly check the difference and return a well format error. Should the comment indicate that it's not known why they would be different?

Reply:
it's a copy of the original comment :wink: 

https://github.com/golang/tools/blob/v0.28.0/go/analysis/checker/checker.go#L360

Reply:
🤦 Oh sorry. I thought it was you since it was a lone diff, should've checked upstream first!